### DISCUSSION TITLE
npm install and test not working on windows (cmd.exe)

### DISCUSSION THREAD

**Discussion by shadowspawn** [community/community#25817] [Actions]
<p>I am trying GitHub Actions with two different npm projects. Both appear to silently fail to run the tests on Windows, but run as expected on macOS and Ubuntu. The job seems to stop after the npm install. I have tried npm install and npm ci, I have tried adding –loglevel=error in case warnings in the output were the issue, I have tried npm install || exit 0.  I do not have a lot of experience with Windows scripting or Actions so suggestions welcome. </p>
<p><a href="https://github.com/shadowspawn/forest-arborist/runs/222728758" rel="nofollow noopener">https://github.com/shadowspawn/forest-arborist/runs/222728758</a></p>
<p>Note the install runs, but apparently not the subsequent echo or the tests. (The hash warning is from using yaml comment character in the run commands, and can be disregarded.)</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/b5827d287c1a34edfcf3232f5d40edef2dc145e4.png" data-download-href="/uploads/short-url/pTHZAlSvewYkXx9XT3v1o6hSoGU.png?dl=1" title="Screen Shot 2019-09-15 at 23.12.35.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/15719847/181093601-d2721559-7b3d-4791-849d-a416058994dc.png" alt="Screen Shot 2019-09-15 at 23.12.35.png" data-base62-sha1="pTHZAlSvewYkXx9XT3v1o6hSoGU" width="563" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/b5827d287c1a34edfcf3232f5d40edef2dc145e4_2_563x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/b/b5827d287c1a34edfcf3232f5d40edef2dc145e4_2_844x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/b/b5827d287c1a34edfcf3232f5d40edef2dc145e4_2_1126x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/b5827d287c1a34edfcf3232f5d40edef2dc145e4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-09-15 at 23.12.35.png</span><span class="informations">1356×1204 101 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by shadowspawn** [comment]
<p>I tried specifying</p>
<pre><code>shell: powershell
</code></pre>
<p> and that got me past the install. The npm test  failed in an opaque way without expected output, but at least I have a workaround for the first blocker of continuing past the npm install.</p>

**Comment by shadowspawn** [comment]
<p>Discovered cmd.exe advice to prefix calls with <code>call</code> if running a script rather than an executable. Switching back to cmd.exe and prefxing all npm commands with call and…</p>
<p>Success! Well, somewhat. The GitHub Action did run the install and then the tests. But ignored the tests failing. Progress…</p>
<pre><code>call npm install
call npm run test
</code></pre>

**Comment by shadowspawn** [comment, marked as answer]
<p>I have things working properly on Windows after switching away from multiline run command. e.g.</p>
<pre><code>- name: npm install
      run: npm ci
    - name: test
      run: npm run test
</code></pre>
<p>Don’t know if the other problems I hit were bugs or user error, but I am unblocked.</p>

**Comment by jbolda** [comment]
<p>I am seeing the same issue myself with your resolution being the only I’ve figured out as well. Unfortunately that becomes painful in larger cases with os matrixes.</p>

**Comment by aaronlippold** [comment]
<p>Still seeing issues with windows-2016 as well </p>

**Comment by codealongwith** [comment]
<p> just check using git bash whether npm -v is exeuted or not.</p>
<p>If npm is working in git bash , then its not path problem…your path reference is ok.</p>
<p>go to environment variable-&gt; system vaiable</p>
<p>just check ComSpec : %SystemRoot%\system32\cmd.exe  is correct or not</p>
<p>in many cases if you have a slash in the end it will not work .for example </p>
<p>ComSpec : %SystemRoot%\system32\cmd.exe\</p>
<p>in such case remove the slash if it exists at the end.</p>
<p>Otherwise check where does your cmd.exe locate and then put that path in ComSpec</p>
<p>Usually it resides at c://Windows/System32/cmd.exe</p>
<p>I hope it works. </p>
<p>-Sudip</p>

**Comment by shadowspawn** [comment]
<p><a class="mention" href="/codealongwith">@codealongwith</a> </p>
<p>Thanks for the reply, but I don’t think you understand the context. The commands are running in a GitHub Action in the cloud. The npm install worked, but the following commands did not run.</p>
